SparkPost/2web2ui
# AC-1865 - (UI) Incorporate global web analytics tracking for Modal component open and close events

AC-1865 - (UI) Incorporate global web analytics tracking for <Modal /> component open and close events

### What Changed

- Added segment tracking for when Modal is opened and closed
- Also added sending Modal name along with these events
      - For Legacy Modals added title prop to them.
      - For new Modals this name is parsed from Modal.Header.
- Added cypress tests and units test coverage for the change.

### How To Test
 Check if Opening and Closing of Modals send correct event and name to Segment,  it is not necessary to check each and every Modal in the app. Covering a few Modal of each type, namely a few Legacy Modal and a few new Modals should suffice.
- Check if All Legacy Modals have a title.
      - For the Legacy Modal, I have noted with each file, how to get to these locations easily. Check comments below on each file.
- Check when opening a new Modal, name gets correctly parsed from the header.
       - For new Modal, these are the only location it is present at, it would suffice to check one on the dashboard which open up by clicking `Change Report`. Worst that could happen in this case is Modal.Header is not set by developer when they add a new Modal, in that case, this should show warning in console, and name won't be tracked but I have also added the location to the segment event so that data isn't completely useless and we know on which location was this Modal opened.
<img width="588" alt="Screen Shot 2021-03-15 at 5 10 57 PM" src="https://user-images.githubusercontent.com/4179337/111221886-76934280-85b1-11eb-8a07-509de1311a6e.png">


### To Do

- [x] Add test for name parsing in the Modal and Legacy components.
- [x] Add supporting unit test to TrackModal.
